tiledb/ml/models/tensorflow_keras.py

```diff
@@ -80,76 +77,77 @@ def load(
+                    model_array_results["model_weights"].item(0)
+                )
+                model.set_weights(model_weights)
 
-            # Compile model.
-            model.compile(
-                **saving_utils.compile_args_from_training_config(
-                    training_config, custom_objects
+            if compile_model:
+                optimizer_weights = pickle.loads(
+                    model_array_results["optimizer_weights"].item(0)
                 )
-            )
-            saving_utils.try_build_compiled_arguments(model)
-
-            # Set optimizer weights.
-            if optimizer_weights:
-                try:
-                    model.optimizer._create_all_weights(model.trainable_variables)
-                except (NotImplementedError, AttributeError):
-                    logging.warning(
-                        "Error when creating the weights of optimizer {}, making it "
-                        "impossible to restore the saved optimizer state. As a result, "
-                        "your model is starting with a freshly initialized optimizer."
-                    )
+                training_config = json.loads(model_array.meta["training_config"])
 
-                try:
-                    model.optimizer.set_weights(optimizer_weights)
-                except ValueError:
-                    logging.warning(
-                        "Error in loading the saved optimizer "
-                        "state. As a result, your model is "
-                        "starting with a freshly initialized "
-                        "optimizer."
+                # Compile model.
+                model.compile(
+                    **saving_utils.compile_args_from_training_config(
+                        training_config, custom_objects
                     )
-        return model
+                )
+                saving_utils.try_build_compiled_arguments(model)
+
+                # Set optimizer weights.
+                if optimizer_weights:
+                    try:
+                        model.optimizer._create_all_weights(model.trainable_variables)
+                    except (NotImplementedError, AttributeError):
+                        logging.warning(
+                            "Error when creating the weights of optimizer {}, making it "
+                            "impossible to restore the saved optimizer state. As a result, "
+                            "your model is starting with a freshly initialized optimizer."
+                        )
+
+                    try:
+                        model.optimizer.set_weights(optimizer_weights)
+                    except ValueError:
+                        logging.warning(
+                            "Error in loading the saved optimizer "
+                            "state. As a result, your model is "
+                            "starting with a freshly initialized "
+                            "optimizer."
```

Comment:
[styling] Just make it a single string and let `black` worry about formatting, this shouldn't take 4 lines.